cookbook/security/impersonating_user.rst

```diff
@@ -151,3 +151,49 @@ setting:
                 ),
             ),
         ));
+        
+Events
+------
+
+The firewall dispatches the ``security.switch_user`` event right after the impersonation
+completed. The ``SwitchUserEvent`` is passed to the listener, based on which you are able
```

Comment:
I agree, will look for an example in another article. Do you have a page where all available roles are documented?